I found one issue in the new manual-cancel coverage.

`TestPurgeMaterializedViewLogManualCancelAfterPartialSuccess` is racy and can fail on the current PR head. After the test writes `CANCEL_REQUESTED_AT`, it immediately disables `pausePurgeMaterializedViewLogAfterDeleteBatch`, but the purge task only observes that cancel request through the async task monitor poll. If the purge goroutine resumes before the next monitor poll, it can finish the remaining work, commit successfully, and return `nil` instead of the expected manual-cancel error.

I reproduced this locally on `332fe9223d7ddd229231ee3645969fe49220872e` with:

```bash
bash -lc 'make failpoint-enable && go test -run "TestPurgeMaterializedViewLog(DeleteErrorAfterPartialSuccess|ManualCancelAfterPartialSuccess)" -tags=intest -count=1 ./pkg/executor/test/ddl; ret=$?; make failpoint-disable; exit $ret'
```

The failure was:

```text
mview_log_ddl_test.go:1463: Error: An error is expected but got nil.
```

The fix could be either to make the test wait until the monitor has actually observed the cancel request before releasing the pause, or, if the intended product semantics are that cancel should be honored at each delete-batch boundary, add a synchronous cancel check after each batch instead of relying only on the async monitor poll.

--- a/pkg/executor/materialized_view.go
+++ b/pkg/executor/materialized_view.go
@@ -262,8 +262,8 @@ func startMVTaskMonitor(
 					zap.Error(err),
 				)
 			} else if requested {
-				failpoint.InjectCall("mvTaskMonitorCancelRequested", monitorName)
 				taskCancelController.requestManualCancelByRequester(requester)
+				failpoint.InjectCall("mvTaskMonitorCancelRequested", monitorName)
 				return
 			}
 

--- a/pkg/executor/materialized_view.go
+++ b/pkg/executor/materialized_view.go
@@ -262,6 +262,7 @@ func startMVTaskMonitor(
 					zap.Error(err),
 				)
 			} else if requested {
+				failpoint.InjectCall("mvTaskMonitorCancelRequested", monitorName)
 				taskCancelController.requestManualCancelByRequester(requester)
 				return
 			}

--- a/pkg/executor/test/ddl/mview_log_ddl_test.go
+++ b/pkg/executor/test/ddl/mview_log_ddl_test.go
@@ -1444,6 +1444,25 @@ func TestPurgeMaterializedViewLogManualCancelAfterPartialSuccess(t *testing.T) {
 		return fmt.Sprint(rows[0][0]) == "1"
 	}, 10*time.Second, 100*time.Millisecond)
 
+	purgeJobID := fmt.Sprint(tkObserver.MustQuery(fmt.Sprintf(
+		"select PURGE_JOB_ID from mysql.tidb_mlog_purge_hist where MLOG_ID = %d and PURGE_STATUS = 'running' limit 1",
+		mlogID,
+	)).Rows()[0][0])
+	cancelObservedCh := make(chan struct{}, 1)
+	cancelObservedFailpoint := "github.com/pingcap/tidb/pkg/executor/mvTaskMonitorCancelRequested"
+	expectedMonitorName := fmt.Sprintf("mlog-purge-%s", purgeJobID)
+	require.NoError(t, failpoint.EnableCall(cancelObservedFailpoint, func(monitorName string) {
+		if monitorName == expectedMonitorName {
+			select {
+			case cancelObservedCh <- struct{}{}:
+			default:
+			}
+		}
+	}))
+	defer func() {
+		require.NoError(t, failpoint.Disable(cancelObservedFailpoint))
+	}()
+
 	requester := "'partial_cancel_req'@'stage-d'"
 	tkObserver.MustExec(
 		`UPDATE mysql.tidb_mlog_purge_hist
@@ -1455,6 +1474,12 @@ WHERE MLOG_ID = ?
 		requester,
 		mlogID,
 	)
+
+	select {
+	case <-cancelObservedCh:
+	case <-time.After(10 * time.Second):
+		t.Fatal("timed out waiting for purge task monitor to observe manual cancel request")
+	}
 
 	require.NoError(t, failpoint.Disable(pauseFailpoint))
 	paused = false

